pkg/batcher/createfleet.go

```diff
@@ -19,17 +19,21 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/service/ec2"
-	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
 type CreateFleetBatcher struct {
 	batcher *Batcher[ec2.CreateFleetInput, ec2.CreateFleetOutput]
 }
 
-func NewCreateFleetBatcher(ctx context.Context, ec2api ec2iface.EC2API) *CreateFleetBatcher {
+type EC2API interface {
```

Comment:
I wonder if we create a generic interface for all of Karpenter. Right now we are creating one per-package, which gives us more flexibility but is also more confusing IMO